packages/spec/src/components/accordion/rules.js

```diff
@@ -0,0 +1,343 @@
+        pressSpace(document.activeElement);
+
+        if (headers[i].getAttribute('aria-expanded') === 'false') {
+          return new ValidationError([
+            'Expected aria-expanded to be true after the heading is clicked',
+          ]);
+        }
+      }
+    },
+  },
+
+  {
+    id: 'accordion.header.interaction.collapse.keyboard.space',
+    description:
+      'When focus is on the accordion header for a expanded panel ' +
+      'then pressing Space will collapse the associated panel',
+    reference: 'https://www.w3.org/TR/wai-aria-practices/#accordion',
+    level: 'error',
+    context: defaultContext,
+    validate(root, context) {
+      const headers = root.querySelectorAll('.bx--accordion__heading');
+
+      for (let i = 0; i < context.children.length; i++) {
+        pressTab(root);
+        pressSpace(document.activeElement);
+        pressSpace(document.activeElement);
+
+        if (headers[i].getAttribute('aria-expanded') === 'true') {
+          return new ValidationError([
+            'Expected aria-expanded to be false after the heading is collapsed',
+          ]);
+        }
+      }
+    },
+  },
+
+  {
+    id: 'accordion.header.interaction.move.keyboard.tab',
+    description:
+      'Moves focus to the next focusable element; all focusable ' +
+      'elements in the accordion are included in the page Tab sequence.',
+    reference: 'https://www.w3.org/TR/wai-aria-practices/#accordion',
+    level: 'error',
+    context: defaultContext,
+    validate(root, context) {
+      const headers = root.querySelectorAll('.bx--accordion__heading');
+
+      for (let i = 0; i < context.children.length; i++) {
+        pressTab(document.body);
+        if (document.activeElement !== headers[i]) {
+          return new ValidationError([
+            'Expected elements in accordion to follow page Tab sequence.',
+          ]);
+        }
+      }
+    },
+  },
+
+  {
+    id: 'accordion.interaction.movement.keyboard.shift+tab',
+    description:
+      'Moves focus to the previous focusable element; all focusable elements ' +
+      'in the accordion are included in the page Tab sequence.',
+    reference: 'https://www.w3.org/TR/wai-aria-practices/#accordion',
+    level: 'error',
+    context: defaultContext,
+    validate(root, context) {
+      const headers = root.querySelectorAll('.bx--accordion__heading');
+
+      for (let i = context.children.length - 1; i >= 0; i--) {
+        pressShiftTab();
+        if (document.activeElement !== headers[i]) {
+          return new ValidationError([
+            'Expected elements in accordion to follow page Tab sequence.',
+          ]);
+        }
+      }
+    },
+  },
+
+  {
+    id: 'accordion.interaction.movement.keyboard.down-arrow',
+    description:
+      'If focus is on an accordion header, moves focus to ' +
+      'the next accordion header. If focus is on the last accordion header, ' +
+      'either does nothing or moves focus to the first accordion header.',
```

Comment:
What is the actual requirement around the behavior of the accordion, it seems like there are two different behaviors but either is allowed. Perhaps setting it up with a _preferred_ way and then throw a warning informing the developer that they are deviating from Carbon but would still meet a11y requirements.

Reply:
@elizabethsjudd still doing a pass on the descriptions, the focus on this was writing rules and figuring out what felt awkward 👍 In this case, the description was pulled from WCAG as a placeholder.